### PR TITLE
update deno.land/std usage to jsr.io/@std

### PR DESCRIPTION
**Pull request by mattvr**
I was following a docs example and saw there was some slightly outdated information. As I pulled on this thread, I found many other places like this with a similar root cause.

I assume that most official https://deno.land pointers should now go to https://jsr.io or https://docs.deno.com, so that's the spirit of this change. 

## Changes

- Updated all examples referencing `https://deno.land/std` with their corresponding `https://jsr.io/@std` docs by updating the `$std` replacement logic.
  - **NOTE:** This also removes version pinning in std examples (i.e. `@std/http` not `@std/http@0.224.0`). Since it doesn't seem like there's a global std version any more, and many examples point to latest, this seemed OK – but let me know if not.
- Updated all instances of https://deno.land/api?s=[...] to point to corresponding https://docs.deno.com/api/[...] paths.
- Updated file-server installation instructions for jsr.io. The module now installs as `file-server` not `file_server` and requires an additional `--allow-sys` flag.
- Updated Deno 1.x -> 2.x migration guide to link to correct type for TlsOptions.
- Removed `$STD_VERSION` as it was only used in one place.

## Testing & validation

- Manually tested all updated links/examples to ensure they resolve correctly.
- Ran `deno task serve` and ensured the site still builds and runs correctly.

## Other notes
- Did not update https://deno.land/std imports and documentation involved with developing this docs repo itself.
- minor: I noticed https://docs.deno.com/api/deno/~/Deno occasionally doesn't render correctly when testing.
- minor: I noticed jsr.io docs paths could potentially be cleaner or more idiomatic, i.e. right now it's `[pkg]/doc/[path]/~/[symbol]`. It could instead be `[pkg]/doc/[path]/[symbol]` or `/doc/[pkg]/[path]/[symbol]`.

### CODE REVIEW

File: by-example/permissions.ts [hunk 1]
```diff
@@ -42,7 +42,7 @@ console.log(readStatus.state);
 // In the case that we no longer need a permission, it is also possible
 // to revoke a process's access to that permission. This is useful when
 // a process starts running untrusted code.
-import { assert } from "$std/assert/assert.ts";
+import { assert } from "jsr:@std/assert/assert.ts";
```
Comment (lucacasonato) [inline by-example/permissions.ts:45]:
```suggestion
import { assert } from "jsr:@std/assert@1/assert.ts";
```

File: by-example/byte-manipulation.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
  * @difficulty beginner
  * @tags cli
  * @run <url>
- * @resource {$std/bytes} Doc: std/bytes
+ * @resource {$std/bytes} Doc: @std/bytes
```
Comment (lucacasonato) [inline by-example/byte-manipulation.ts:6]:
```suggestion
 * @resource {https://jsr.io/@std/bytes/doc} Doc: @std/bytes
```

File: README.md
Comment (lucacasonato) [inline README.md:1]: The entire section in `Including version numbers in code and content` needs to be updated to not mention the standard library anymore.

File: by-example/command-line-arguments.ts [hunk 3]
```diff
@@ -3,8 +3,8 @@
  * @difficulty beginner
  * @tags cli
  * @run <url> Deno Sushi --help --version=1.0.0 --no-color
- * @resource {https://deno.land/api?s=Deno.args} Doc: Deno.args
- * @resource {$std/cli/parse_args.ts} Doc: std/cli
+ * @resource {https://docs.deno.com/api/deno/~/Deno.args} Doc: Deno.args
+ * @resource {$std/cli/parse-args} Doc: @std/cli
```
Comment (lucacasonato) [inline by-example/command-line-arguments.ts:7]:
```suggestion
 * @resource {https://jsr.io/@std/cli/doc/parse-args/~} Doc: @std/cli
```

File: by-example/http-server-files.ts [hunk 4]
```diff
@@ -3,8 +3,8 @@
  * @difficulty intermediate
  * @tags cli, deploy
  * @run --allow-net --allow-read <url>
- * @resource {https://deno.land/api?s=Deno.serve} Doc: Deno.serve
- * @resource {$std/http/file_server.ts} Doc: std/http/file_server
+ * @resource {https://docs.deno.com/api/deno/~/Deno.serve} Doc: Deno.serve
+ * @resource {$std/http/file-server} Doc: @std/http/file-server
```
Comment (lucacasonato) [inline by-example/http-server-files.ts:7]:
```suggestion
 * @resource {https://jsr.io/@std/http#file-server} Doc: @std/http/file-server
```

File: by-example/path-operations.ts [hunk 5]
```diff
@@ -3,8 +3,8 @@
  * @difficulty beginner
  * @tags cli
  * @run --allow-read <url>
- * @resource {$std/path} Deno: std/path
- * @resource {https://deno.land/api?s=Deno.cwd} Deno: Deno.cwd
+ * @resource {$std/path} Deno: @std/path
```
Comment (lucacasonato) [inline by-example/path-operations.ts:6]:
```suggestion
 * @resource {https://jsr.io/@std/path/doc} Doc: @std/path
```

File: by-example/ulid.ts [hunk 6]
```diff
@@ -28,13 +28,13 @@ console.log(ulid(timestamp));
 console.log(ulid(timestamp));
 
 // Given a ULID, you can get the timestamp back out
-import { decodeTime } from "$std/ulid/mod.ts";
+import { decodeTime } from "jsr:@std/ulid/mod.ts";
```
Comment (lucacasonato) [inline by-example/ulid.ts:31]:
```suggestion
import { decodeTime } from "jsr:@std/ulid@0.224/mod.ts";
```

File: by-example/ulid.ts [hunk 7]
```diff
@@ -28,13 +28,13 @@ console.log(ulid(timestamp));
...
 const myULID = ulid();
 console.log(decodeTime(myULID));
 
 // Optionally, if you're not on a distributed system and want monotonic ULIDs,
 // you can use the monotonic ULID generator instead.
-import { monotonicUlid } from "$std/ulid/mod.ts";
+import { monotonicUlid } from "jsr:@std/ulid/mod.ts";
```
Comment (lucacasonato) [inline by-example/ulid.ts:37]:
```suggestion
import { monotonicUlid } from "jsr:@std/ulid@0.224/mod.ts";
```

File: by-example/uuids.ts [hunk 8]
```diff
@@ -4,7 +4,7 @@
  * @tags cli, deploy, web
  * @run <url>
  * @resource {https://developer.mozilla.org/en-US/docs/Web/API/Crypto/randomUUID} MDN: crypto.randomUUID
- * @resource {$std/uuid/mod.ts} Doc: std/uuid
+ * @resource {$std/uuid} Doc: @std/uuid
```
Comment (lucacasonato) [inline by-example/uuids.ts:7]:
```suggestion
 * @resource {https://jsr.io/@std/uuid/doc} Doc: @std/uuid
```

File: by-example/walking-directories.ts [hunk 9]
```diff
@@ -3,8 +3,8 @@
  * @difficulty beginner
  * @tags cli
  * @run --allow-read <url>
- * @resource {https://deno.land/api?s=Deno.readDir} Doc: Deno.readDir
- * @resource {$std/fs/walk.ts} Doc: std/walk
+ * @resource {https://docs.deno.com/api/deno/~/Deno.readDir} Doc: Deno.readDir
+ * @resource {$std/fs/walk} Doc: @std/walk
```
Comment (lucacasonato) [inline by-example/walking-directories.ts:7]:
```suggestion
 * @resource {https://jsr.io/@std/fs/doc/walk/~} Doc: @std/fs/walk
```

File: by-example/watching-files.ts [hunk 10]
```diff
@@ -3,9 +3,9 @@
  * @difficulty beginner
  * @tags cli
  * @run --allow-read <url>
- * @resource {https://deno.land/api?s=Deno.watchFs} Doc: Deno.watchFs
+ * @resource {https://docs.deno.com/api/deno/~/Deno.watchFs} Doc: Deno.watchFs
  * @resource {https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for-await...of} MDN: for await of
- * @resource {$std/async/debounce.ts} Doc: std/debounce
+ * @resource {$std/async/debounce} Doc: @std/debounce
```
Comment (lucacasonato) [inline by-example/watching-files.ts:8]:
```suggestion
 * @resource {https://jsr.io/@std/async/doc/debounce/~} Doc: @std/async/debounce
```

File: by-example/writing-tests.ts [hunk 11]
```diff
@@ -3,8 +3,8 @@
  * @difficulty beginner
  * @tags cli
  * @run deno test --allow-read --allow-write <url>
- * @resource {https://deno.land/api?s=Deno.test} Doc: Deno.test
- * @resource {$std/testing/asserts.ts} Doc: std/testing/asserts
+ * @resource {https://docs.deno.com/api/deno/~/Deno.test} Doc: Deno.test
+ * @resource {$std/assert} Doc: @std/asserts
```
Comment (lucacasonato) [inline by-example/writing-tests.ts:7]:
```suggestion
 * @resource {https://jsr.io/@std/assert/doc} Doc: @std/assert
```

File: runtime/tutorials/file_server.md [hunk 12]
```diff
@@ -64,24 +64,24 @@ async function handleHttp(conn: Deno.Conn) {
...
 to write your own.
 
 To use it, first install the remote script to your local file system. This will
 install the script to the Deno installation root's bin directory, e.g.
-`/home/alice/.deno/bin/file_server`.
+`/home/alice/.deno/bin/file-server`.
 
 ```shell
-deno install --allow-net --allow-read https://deno.land/std/http/file_server.ts
+deno install --allow-net --allow-read --allow-sys jsr:@std/http/file-server
```
Comment (lucacasonato) [inline runtime/tutorials/file_server.md:75]:
```suggestion
deno install --allow-net --allow-read --allow-sys jsr:@std/http@1/file-server
```

File: runtime/tutorials/file_server.md [hunk 13]
```diff
@@ -99,7 +99,7 @@ Example output:
 Deno File Server
     Serves a local directory in HTTP.
   INSTALL:
-    deno install --allow-net --allow-read https://deno.land/std/http/file_server.ts
+    deno install --allow-net --allow-read jsr:@std/http/file_server
```
Comment (lucacasonato) [inline runtime/tutorials/file_server.md:102]:
```suggestion
    deno install --allow-net --allow-read jsr:@std/http@1/file_server
```

**Review by lucacasonato — COMMENTED** [review]
We should also add version constraints to all `jsr:@std` imports.

**Comment by mattvr** [comment]
Thanks for the review + pushing this along @lucacasonato, I'm out on vacation but let me know if there's anything you'd like help with.

**Review by thisisjofrank — APPROVED** [review]
This is an amazing amount of work. thank you so much.

**Pull request merged**